### ISSUE TITLE
insert does not take a function

### ISSUE THREAD

**Issue by mren** [snd/mohair#2]
Use Case:

On Duplicate Key Update does not work

``` sql
INSERT INTO table (id, value) VALUES (?, ?) ON DUPLICATE KEY UPDATE value=?
```

what do you think of this syntax:

``` coffeescript
m = mohair()
m.insert 'table', {id: id, value: value}, ->
  m.onDuplicateKeyUpdate {value: value}
```

**Comment by snd** [comment]
done in 0d2cd5c08c1f9adc0f13c3d22f6b0116e658ba9a and available in version 0.1.1

after the `VALUES ...` part in an insert query only `ON DUPLICATE ...` is possible syntax.
thus we can make the command even simpler without loosing power:

``` coffeescript
m.insert 'table', {id: id, value: value}, {value: value}
```

=> 

``` sql
INSERT INTO table (id, value) VALUES (?, ?) ON DUPLICATE KEY UPDATE value = ?;
```

or something like this:

``` coffeescript
m.insert 'table', {id: id, value: value}, {value: -> m.raw 'id'}
```

=> 

``` sql
INSERT INTO table (id, value) VALUES (?, ?) ON DUPLICATE KEY UPDATE value = id;
```

**Comment by mren** [comment]
nice :)
